cdci_data_analysis/analysis/instrument.py

```diff
@@ -128,24 +129,26 @@ def set_pars_from_dic(self, par_dic, verbose=False):
             query_obj = self.get_query_by_name(query_name)
             # loop over the list of parameters for the requested query,
             # but also of the instrument query and source query
-            for par in (query_obj.parameters +
-                        self.instrumet_query.parameters +
-                        self.src_query.parameters):
-                # this is required because in some cases a parameter is set without a name (eg UserCatalog),
-                # or they don't have to set (eg scw_list)
-                # 
-                if par.name is not None and par.name not in params_not_to_be_included:
-                    par.set_from_form(par_dic, verbose=verbose)
-
-                self.logger.info("set_pars_from_dic>> par: %s par.name: %s par.value: %s par_dic[par.name]: %s", par, par.name, par.value, par_dic.get(par.name, None))
-                if par.name == "scw_list":
-                    self.logger.info("set_pars_from_dic>> scw_list is %s", par.value)
-
+            param_list = (query_obj.parameters +
+                          self.instrumet_query.parameters +
+                          self.src_query.parameters)
         else:
-            for _query in self._queries_list:
-                for par in _query.parameters:
-                    if par.name is not None and par.name not in params_not_to_be_included:
-                        par.set_from_form(par_dic, verbose=verbose)
+            param_list = [par for _query in self._queries_list for par in _query.parameters]
+
+        for par in param_list:
+            # this is required because in some cases a parameter is set without a name (eg UserCatalog),
+            # or they don't have to set (eg scw_list)
+            if par.name is not None and par.name not in params_not_to_be_included:
+                # set the value for par in to a default format,
+                # or to a default value if this is not included within the request
+                par.set_value_from_form(par_dic, verbose=verbose)
```

Comment:
I'd have this function return value, and maybe even units (as tuple). This way also `set_value_from_form` will not manipulate its arguments.